### ISSUE TITLE
Development server

### ISSUE THREAD

**Issue by MarcAntoine-Arnaud** [rustwasm/wasm-bindgen#2359]
### Motivation

Wasm-bindgen provide all requirements to generate Wasm application. Including the startup with the `#[wasm_bindgen(start)]` attribute.
It miss a very little portion of code to:
- start a server
- (re-)build the wasm application

Today, it requires an external tool, hot reloading is not provided etc.
It can be with WebPack:
[Paint example](https://github.com/rustwasm/wasm-bindgen/tree/master/examples/paint)
Or using the `wasm-bindgen-cli`
[GFX example](https://github.com/gfx-rs/gfx/tree/master/examples#running-quad-with-webglwebassembly)

One of the paint/complex work is to manage cross target libraries.
For example the Quad example from GFX is a `main` as it can run on all platforms.
But the same code can also run for Web with WebAssembly.
I have tried to manage that in that issue: https://github.com/gfx-rs/gfx/pull/3475
It works using 2 Cargo manifests, requires to duplicate dependencies and do some tricks.

### Proposed Solution

Improve the CLI crate to add a mode to serve the application.
maybe to provide a command like:
`wasm-bindgen --server`
Or maybe add a `server` target.

**Comment by alexcrichton** [comment]
Thanks for the report! This is something I'd be happy to help design and take a PR for, but I don't really know of a great design or how it would fit into `wasm-bindgen` myself. There's already basically a server for `wasm-bindgen-test` so that can probably be reused.

This would be a great feature to have though!

**Comment by MarcAntoine-Arnaud** [comment]
Hello,

I wanna to work on it, @alexcrichton do you have some advices ?
Do you have idea what I need to check first ?

For me it means:
- call `cargo build` command - maybe with watch option
- call wasm-bindgen-cli 
- serve over HTTP (I plan to use https://crates.io/crates/https)

Thank you,
Marc-Antoine

**Comment by alexcrichton** [comment]
Yeah that's the rough shape of ideally what this would look like. I think it'd be good though to make a proposal for the interface for this and how it's expected to be used, though.
